core/runtime/src/main/java/io/quarkus/runtime/annotations/PreStart.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.runtime.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import jakarta.enterprise.util.AnnotationLiteral;
+import jakarta.inject.Qualifier;
+
+@Qualifier
+@Documented
+@Inherited
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface PreStart {
```

Comment:
What's the rationale for this qualifier? It's part of the public API and it's not clear if it's intended to be used by the users or not.

Reply:
It's intended to be used by users. (let me add more comments.

Reply:
> What's the rationale for this qualifier?

You need this qualifier in order to distinguish the `Runnable`s that should be considered _initialization tasks_. I.e. when obtaining the beans that should be executed. Otherwise, all beans that implement `Runnable` would be used.

So I guess that it's a user facing API that needs to be properly documented.

Also I'm not so sure it's a good name. I mean naming is hard and I have no better name in my sleeve...